Source/WebCore/bindings/js/SerializedScriptValue.cpp

```diff
@@ -3378,39 +3373,42 @@ class CloneDeserializer : public CloneBase {
 
     template <typename T> bool readLittleEndian(T& value)
     {
-        if (m_failed || !readLittleEndian(m_ptr, m_end, value)) {
+        if (m_failed || !readLittleEndian(m_data, value)) {
             SERIALIZE_TRACE("FAIL deserialize");
             fail();
             return false;
         }
         return true;
     }
 #if ASSUME_LITTLE_ENDIAN
-    template <typename T> static bool readLittleEndian(const uint8_t*& ptr, const uint8_t* end, T& value)
+    template <typename T> static bool readLittleEndian(std::span<const uint8_t>& span, T& value)
     {
-        if (ptr > end - sizeof(value))
+        if (span.size() < sizeof(value))
             return false;
 
-        if (sizeof(T) == 1)
-            value = *ptr++;
-        else {
-            value = *reinterpret_cast<const T*>(ptr);
-            ptr += sizeof(T);
+        if (sizeof(T) == 1) {
+            value = span[0];
+            span = span.subspan(1);
+        } else {
+            value = reinterpretCastSpanStartTo<const T>(span);
+            span = span.subspan(sizeof(T));
         }
         return true;
     }
 #else
-    template <typename T> static bool readLittleEndian(const uint8_t*& ptr, const uint8_t* end, T& value)
+    template <typename T> static bool readLittleEndian(std::span<const uint8_t>& span, T& value)
     {
-        if (ptr > end - sizeof(value))
+        if (span.size() < sizeof(value))
             return false;
 
-        if (sizeof(T) == 1)
-            value = *ptr++;
-        else {
+        if (sizeof(T) == 1) {
+            value = span[0];
+            span = span.subspan(1);
+        } else {
             value = 0;
-            for (unsigned i = 0; i < sizeof(T); i++)
-                value += ((T)*ptr++) << (i * 8);
+            for (size_t i = 0; i < sizeof(T); ++i)
+                value += ((T)span[i]) << (i * 8);
```

Comment:
Does this need to be a C-style cast? Maybe we can assign to a local variable of type `T` and avoid a cast. Or maybe `static_cast` if that won't work.

Reply:
Looks like static_cast works.